### PR TITLE
Feat/redis

### PR DESCRIPTION
**Pull request by deimosfr**

### CODE REVIEW

File: src/cloud_provider/aws/databases/redis.rs [hunk 1]
```diff
@@ -577,3 +569,25 @@ impl Backup for Redis {
         unimplemented!()
     }
 }
+
+fn get_redis_version(
+    requested_version: &str,
+    is_managed_service: bool,
+) -> Result<String, StringError> {
+    let mut supported_redis_versions = HashMap::with_capacity(2);
+
```
Comment (evoxmusic) [inline src/cloud_provider/aws/databases/redis.rs:578]: I think you should check the input. The version must be 5 or 6 in this case. Other versions should throw an error.

File: src/cloud_provider/aws/databases/utilities.rs [hunk 2]
```diff
@@ -74,3 +83,82 @@ pub fn delete_terraform_tfstate_secret(
         }
     }
 }
+
+pub fn get_version_semver(version: &str) -> Result<VersionsNumber, StringError> {
```
Comment (evoxmusic) [inline src/cloud_provider/aws/databases/utilities.rs:87]: 1/ I'd export this function outside of the Redis struct. 
2/ I'd rename it to `get_semver` or `get_version_number`. `version: &str` already gives details on the expected input.

File: src/cloud_provider/aws/databases/utilities.rs [hunk 3]
```diff
@@ -74,3 +83,82 @@ pub fn delete_terraform_tfstate_secret(
...
+
+    Ok(VersionsNumber {
+        major,
+        minor,
+        patch,
+    })
+}
+
+pub fn get_supported_version_to_use(
+    all_supported_versions: HashMap<&str, &str>,
```
Comment (evoxmusic) [inline src/cloud_provider/aws/databases/utilities.rs:117]: I'd suggest using type alias for hashmap key and value, then we can understand what types are expected?

**Review by evoxmusic — COMMENTED** [review]
Here are my suggestions, and you're probably better to add more tests on the provided version inputs.

File: src/cloud_provider/aws/databases/redis.rs (hunk 1 above)
Comment (deimosfr) [reply src/cloud_provider/aws/databases/redis.rs:578]: It's done by the function called get_supported_version_to_use in this function. If it doesn't match, I throws an error saying:
this {} version is not supported

File: src/cloud_provider/aws/databases/utilities.rs (hunk 2 above)
Comment (deimosfr) [reply src/cloud_provider/aws/databases/utilities.rs:87]: done

File: src/cloud_provider/aws/databases/utilities.rs (hunk 3 above)
Comment (evoxmusic) [reply src/cloud_provider/aws/databases/utilities.rs:117]: @deimosfr ->

```rust
type ProvidedVersion<'a> = &'a str;
type RealVersion<'a> = &'a str;

//....
pub fn get_supported_version_to_use(
all_supported_versions: HashMap<ProvidedVersion<'a>, RealVersion<'a>>,...
)
```

**Pull request merged**